apollographql/embeddable-explorer
# NEBULA-2125: support multipart subs in embedded Sandbox only

## Context

[#proj-realtime](https://apollograph.slack.com/archives/C026H578D6K/p1677288914761019) in slack

[JIRA](https://apollographql.atlassian.net/browse/NEBULA-2125)

[Studio-UI PR](https://github.com/mdg-private/studio-ui/pull/8232)

## What changed

This PR mirrors the[ studio-ui support for http multi part subscriptions](https://github.com/mdg-private/studio-ui/pull/8169)

1. Special cased multi part subscriptions to use `executeSubscription` but send subscription response messages to the embed
2. copied all the changes from sandbox into explorer

**Some things to note**
1. ~I am going to add shared code via an npm package before adding support in the explorer code, I think this is the right way forward~ **Spoke with Chang about this, we are going to make a shared package for these changes after file upload is done. When I get the green check on this PR I will open up another one to support the explorer package.** 

3. I noticed a bunch of messiness when putting up this PR w.r.t the types in studio-ui repo & this repo. I would like to do some cleanup with the types as a quick follow. Added todos, but going to do that as a part of the shared repo push

## How to test this PR

[Follow these instructions](https://apollograph.slack.com/archives/C026H578D6K/p1678190088438199?thread_ts=1678147601.919469&cid=C026H578D6K)

I have no good way to test the `done: true` returned by the server if something goes wrong on the server end [as described here ](https://apollographql.quip.com/mkWRAJfuxa7L/Multipart-subscriptions-protocol-spec#temp:C:YCbae12582e61b54b0fba08784cd). I just set the `if` to true to see if it closed :homer-disappear:

Open `localDevelopmentExample` in [the embeddable-explorer repo on this branch](https://github.com/apollographql/embeddable-explorer/pull/224) 

The localDevelopmentExamples have been updated for testing in their respective packages for multi part subs
